project/jasonchen1221/README.md

```diff
@@ -0,0 +1,82 @@
+# R-Tree Search Engine
+
+
+## Basic Information
+Repository & URL: [jasonchen1221](https://github.com/jasonchen1221) for [R-Tree Search Engine: ](https://github.com/jasonchen1221/RTree-Search-Engine)
+
+R-Tree is a tree data structure used for spatial searching. It was proposed by Antonin Guttman in 1984 as an expansion of B-tree for multi-dimensional data. It may be used to store points or volumetric data in order to perform a spatial query. It's possible to insert new objects or to remove the ones already stored. 
+
+This kind of spatial search occurs frequently in Computer Aided Design (CAD) and Electronic Design Automation(EDA) problems, and also geo-data applications, and therefore it is important to be able to retrieve objects efficiently according to their spatial location.
+
+
+## Problem to Solve
+For this project, I will try to implement the R-Tree search engine to meet the requirements as below,
+	1. It works in two-dimensional space and may index point, line, and surface.
+	2. Allow dynamic update of elements.
+	3. Allow access elements using a serial (integer) identifier.
+	4. Support ranged search of the geometrical entities.
+	5. Try to apply the R-Tree search engine on the EDA problems on hand.
+
+
+## Prospective Users
+The R-Tree search engine is for fast spatial searching. Electronic Design Automation(EDA) problems, map applications, and the problems which have large amount of data on the two-dimensional space with coordinates, shapes, and other information which related to the space may take advantage of it's fast speed of space searching.
+
+
+## System Architecture
+_(Still need to think about and work on this part as I think more about this project and know more about how to plan the system architecture…)_
+
+Basically, I will mainly use C++ to do the implementation of the engine for the reason of the high speed operation of R-Tree. And I will use high level language Python to test and verify the correct operation of the R-Tree search engine.
+
+To meet the requirements as mentioned in the section of Problem to Solve, the system architecture would base on the main class "RTREE". Also I may have some different functions to support the operations like dynamically adding a new object, deleting an object, and querying the information at all times.
```

Comment:
What are the (geometrical) data types the search engine would support?  Points and/or anything else?

Reply:
Could you let me know what do you think about my question here?